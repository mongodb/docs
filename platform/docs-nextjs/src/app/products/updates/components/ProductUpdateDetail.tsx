'use client';
import type { ProductUpdateEntry } from '../services/contentstack';
import Link from 'next/link';

export enum LinkTag {
  Docs = 'Docs',
  Blog = 'Blog',
  Tutorial = 'Tutorial',
  Press = 'Press',
  Web = 'Web',
  GitHub = 'GitHub',
}
import Icon from '@leafygreen-ui/icon';
import { palette } from '@leafygreen-ui/palette';
import { Body, H3 } from '@leafygreen-ui/typography';
import { css, cx } from '@leafygreen-ui/emotion';
import { theme } from '@/styles/theme';
import Card from '@leafygreen-ui/card';
import { useRouter } from 'next/navigation';

const containerStyle = css`
  width: 100vw;
  padding: 64px;
`;

const contentStyle = css`
  max-width: 968px;
  margin: 0 auto;
`;

const headerStyle = css`
  margin-bottom: 40px;
`;

const linkTextStyle = css`
  margin-left: 4px;
  color: ${palette.blue.base};
`;

const linkStyle = css`
  display: flex;
  align-items: center;
  flex-direction: row;
  text-decoration: none;
  font-size: 16px;
  font-weight: 500;
`;

const dateStyle = css`
  font-size: 13px;
  line-height: 20px;
  color: ${palette.gray.dark1};
  margin-bottom: 24px;

  @media ${theme.screenSize.largeAndUp} {
    font-size: 16px;
    line-height: 28px;
  }
`;

const titleStyle = css`
  font-size: 32px;
  line-height: 40px;
  margin-bottom: 12px;
  font-weight: 400;
  color: ${palette.black};

  @media ${theme.screenSize.largeAndUp} {
    font-size: 48px;
    line-height: 64px;
  }
`;

const bodyStyle = css`
  font-size: 16px;
  line-height: 28px;
  color: ${palette.black};
  margin-bottom: 40px;
`;

const relatedContentTitleStyle = css`
  font-size: 24px;
  line-height: 32px;
  margin-bottom: 24px;
  color: ${palette.black};
  font-weight: 500;
`;

const relatedContentGridStyle = css`
  display: grid;
  grid-template-columns: 1fr;
  gap: 16px;

  @media ${theme.screenSize.xLargeAndUp} {
    grid-template-columns: repeat(2, 1fr);
    gap: 24px;
  }
`;

const relatedContentCardStyle = css`
  display: flex;
  flex-direction: column;
  padding: 24px;
  cursor: pointer;
  text-decoration: none;
  transition: all 0.2s ease;
  max-width: 472px;

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
  }
`;

const linkLabelStyle = css`
  display: inline-flex;
  align-items: center;
  gap: 4px;
  margin-bottom: 16px;
  padding: 0 6px;
  width: fit-content;
  border-radius: 6px;
`;

const linkLabelTextStyle = css`
  font-size: 12px;
  line-height: 22px;
  font-weight: 400;
`;

const cardTitleStyle = css`
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  color: ${palette.black};
  margin: 0;
`;

const linkLabelDocsStyle = css`
  background-color: ${palette.blue.light2};
`;

const linkLabelDocsTextStyle = css`
  color: ${palette.blue.dark2};
`;

const linkLabelBlogStyle = css`
  background-color: ${palette.green.light2};
`;

const linkLabelBlogTextStyle = css`
  color: ${palette.green.dark3};
`;

const ProductUpdateDetail = ({ update }: { update: ProductUpdateEntry }) => {
  const router = useRouter();

  const createdAt = update.beamer_created_at || update.created_at;
  const date = new Date(createdAt).toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });

  const getLabel = (label: string) => {
    switch (label) {
      case 'Docs':
        return (
          <div className={cx(linkLabelStyle, linkLabelDocsStyle)}>
            <Icon glyph="Note" color={palette.blue.dark2} />
            <Body className={cx(linkLabelTextStyle, linkLabelDocsTextStyle)}>{label}</Body>
          </div>
        );
      case 'Blog':
        return (
          <div className={cx(linkLabelStyle, linkLabelBlogStyle)}>
            <Icon glyph="SMS" color={palette.green.dark2} />
            <Body className={cx(linkLabelTextStyle, linkLabelBlogTextStyle)}>{label}</Body>
          </div>
        );
      default:
        return (
          <div className={cx(linkLabelStyle, linkLabelDocsStyle)}>
            <Body className={cx(linkLabelTextStyle, linkLabelDocsTextStyle)}>{label}</Body>
          </div>
        );
    }
  };

  return (
    <div className={cx(containerStyle)}>
      <div className={cx(contentStyle)}>
        <div className={cx(headerStyle)}>
          <Link href="/products/updates" className={cx(linkStyle)}>
            <Icon glyph="ArrowLeft" className={cx(linkTextStyle)} />
            <Body className={cx(linkTextStyle)}>See all updates</Body>
          </Link>
        </div>

        <div>
          <H3 className={cx(titleStyle)}>{update.title}</H3>
          <Body className={cx(dateStyle)}>{date}</Body>
        </div>

        <Body className={cx(bodyStyle)}>{update.multi_line || 'No content available'}</Body>
        {update.link_with_label && update.link_with_label.length > 0 && (
          <>
            <H3 className={cx(relatedContentTitleStyle)}>Related Content</H3>
            <div className={cx(relatedContentGridStyle)}>
              {update.link_with_label.map((link) => {
                const label = link.label;
                return (
                  <Card
                    key={link.link.href}
                    className={cx(relatedContentCardStyle)}
                    onClick={() => router.push(link.link.href)}
                  >
                    {getLabel(label)}

                    <Body className={cx(cardTitleStyle)}>{link.link.title}</Body>
                  </Card>
                );
              })}
            </div>
          </>
        )}
      </div>
    </div>
  );
};

export default ProductUpdateDetail;
