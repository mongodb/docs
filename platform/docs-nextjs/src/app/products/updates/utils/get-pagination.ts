/**
 *
 * @param array - flatten array
 * @param chunkSize - max size of each array
 * @returns [[{}, {}], [{}]]
 * @description Takes an flatten array and creates one level nested array of objects based on the chunkSize. Used for pagination data.
 */
export const getPagination = <T>(array: T[], chunkSize: number): T[][] => {
  const chunks: T[][] = [];
  if (!array.length) {
    return [[]]; // default to one empty nested array
  }
  for (let i = 0; i < array.length; i += chunkSize) {
    chunks.push(array.slice(i, i + chunkSize));
  }
  return chunks;
};
