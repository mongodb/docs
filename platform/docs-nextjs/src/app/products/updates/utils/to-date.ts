export const normalizeDate = (date: string) => {
  // Check if the date string has the correct format 'YYYY-MM-DD'
  const dateToNormalize = /^(\d{4})-(\d{2})-(\d{2})$/.exec(date)?.slice(1); // excludes the full date in the first index i.e. "2025-05-27"
  if (dateToNormalize) {
    // This is needed because, JavaScript’s Date constructor uses zero-based months (January = 0 … December = 11).
    return new Date(Number(dateToNormalize[0]), Number(dateToNormalize[1]) - 1, Number(dateToNormalize[2]));
  }
  return new Date(date);
};
