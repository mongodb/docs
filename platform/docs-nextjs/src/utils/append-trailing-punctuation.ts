import { ASTNode, ReferenceNode, RefRoleNode } from "@/types/ast";
import { isReferenceNode, isRefRoleNode, isTextNode } from "@/types/ast-utils";

// Append dangling punctuation to preceding Link's text
export const appendTrailingPunctuation = (nodes: ASTNode[]) => {
  const truncatedNodes = [];

  for (let i = 0; i < nodes.length; i++) {
    const currNode = nodes[i];
    const nextNode = nodes[i + 1];
    const hasDanglingSibling =
      isTextNode(nextNode) &&
      nextNode.value.length === 1 &&
      !nextNode.value.match(/\s/);
    // TODO: if issues arise, instead of matching for non-whitespace, we might consider checking FOR certain singular punctuation
    // ie: regex match for  .?!,:;)*"']}`

    if ((isReferenceNode(currNode) || isRefRoleNode(currNode)) && hasDanglingSibling) {
      const copyOfNodeWithDeepCopiedChildren = deepCopyNodesChildren(currNode);
      copyOfNodeWithDeepCopiedChildren.children.push(nextNode);
      truncatedNodes.push(copyOfNodeWithDeepCopiedChildren);
      i++;
      continue;
    }
    truncatedNodes.push(currNode);
  }
  return truncatedNodes;
};

// Make a copy of node with a deep copy of the new child node
function deepCopyNodesChildren(refNode: ReferenceNode | RefRoleNode) {
  if (!refNode.children || !refNode.children.length) return { ...refNode };
  const copyOfChildren = [...refNode.children];

  const copyOfNode = {
    ...refNode,
    children: copyOfChildren,
  };
  return copyOfNode;
}
