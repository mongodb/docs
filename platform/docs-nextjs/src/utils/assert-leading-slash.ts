const assertLeadingSlash = (url: string) => {
  if (url && url.match(/^\//)) {
    return url;
  }
  return `/${url}`;
};

export { assertLeadingSlash };
