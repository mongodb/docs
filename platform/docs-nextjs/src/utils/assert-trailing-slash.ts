export const assertTrailingSlash = (url: string) => {
  if (url && url.match(/\/$/)) {
    return url;
  }
  return `${url}/`;
};
