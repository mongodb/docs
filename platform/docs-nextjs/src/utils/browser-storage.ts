import { isBrowser } from './is-browser';

const isValidStorage = isBrowser;

export const setLocalValue = (key: string, value: unknown) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      localStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting localStorage value');
  }
};

export const getLocalValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting localStorage value');
    return undefined;
  }
};

export const setSessionValue = (key: string, value: string) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      sessionStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting session value');
  }
};

export const getSessionValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting session value');
    return undefined;
  }
};
