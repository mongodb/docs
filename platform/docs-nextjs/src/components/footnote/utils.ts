import type { ASTNode } from '@/types/ast';
import type { FootnoteNode, FootnoteReferenceNode } from '@/types/ast';
import { findAllKeyValuePairs } from '@/utils/find-all-key-value-pairs';
import type { FootnoteType } from './footnote-context';

export const getFootnotes = (nodes: ASTNode[]) => {
  const footnotes = findAllKeyValuePairs(nodes, 'type', 'footnote') as FootnoteNode[];
  const footnoteReferences = findAllKeyValuePairs(nodes, 'type', 'footnote_reference') as FootnoteReferenceNode[];
  const numAnonRefs = footnoteReferences.filter(
    (node) => !Object.prototype.hasOwnProperty.call(node, 'refname'),
  ).length;
  // We label our footnotes by their index, regardless of their names to
  // circumvent cases such as [[1], [#], [2], ...]
  return footnotes.reduce<Record<string, FootnoteType>>((map, footnote, index) => {
    if (footnote.name) {
      // Find references associated with a named footnote

      map[footnote.name] = {
        label: index + 1,
        references: getNamedFootnoteReferences(footnoteReferences, footnote.name),
      };
    } else {
      // Find references associated with an anonymous footnote
      // Replace potentially broken anonymous footnote ids
      footnote.id = `${index + 1}`;

      map[footnote.id] = {
        label: index + 1,
        references: getAnonymousFootnoteReferences(index, numAnonRefs),
      };
    }
    return map;
  }, {});
};

// Find all footnote_reference node IDs associated with a given footnote by
// that footnote's refname
const getNamedFootnoteReferences = (footnoteReferences: FootnoteReferenceNode[], refname: string) => {
  return footnoteReferences.filter((node) => node.refname === refname).map((node) => node.id);
};

// They are used infrequently, but here we match an anonymous footnote to its reference.
// The nth footnote on a page is associated with the nth reference on the page. Since
// anon footnotes and footnote references are anonymous, we assume a 1:1 pairing, and
// have no need to query nodes. If there are more anonymous footnotes than references,
// we may return an empty array
const getAnonymousFootnoteReferences = (index: number, numAnonRefs: number) => {
  return index > numAnonRefs ? [] : [`id${index + 1}`];
};
