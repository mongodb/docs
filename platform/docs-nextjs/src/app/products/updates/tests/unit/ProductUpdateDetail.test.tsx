import { render } from '@testing-library/react';
import { useRouter } from 'next/navigation';
import ProductUpdateDetail from '../../components/ProductUpdateDetail';

import { mockedUpdate } from '../utils/mock-update';

describe('ProductUpdateDetail component', () => {
  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({ push: jest.fn() });
  });
  it('renders correctly', () => {
    const wrapper = render(<ProductUpdateDetail update={mockedUpdate} />);
    const mainUpdateTitle = 'MongoDB SQL Interface for Enterprise Advanced Now Generally Available';
    expect(wrapper.container.querySelector('a[href="/products/updates')).toBeInTheDocument();
    expect(wrapper.container.querySelector('h3')).toHaveTextContent(mainUpdateTitle);
    // Let us check for the related content as well
    expect(wrapper.getByText('Related Content')).toBeInTheDocument();
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
