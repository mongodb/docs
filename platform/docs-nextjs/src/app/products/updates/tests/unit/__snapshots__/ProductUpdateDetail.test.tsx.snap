// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductUpdateDetail component renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100vw;
  padding: 64px;
}

.emotion-1 {
  max-width: 968px;
  margin: 0 auto;
}

.emotion-2 {
  margin-bottom: 40px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-weight: 500;
}

.emotion-4 {
  margin-left: 4px;
  color: #016BF8;
}

.emotion-5 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  margin-left: 4px;
  color: #016BF8;
}

.emotion-5 strong,
.emotion-5 b {
  font-weight: 600;
}

.emotion-6 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  color: #001E2B;
  line-height: 40px;
  margin-bottom: 12px;
  font-weight: 400;
  color: #001E2B;
  font-family: 'MongoDB Value Serif';
}

@media not all and (max-width: 1024px) {
  .emotion-6 {
    font-size: 48px;
    line-height: 64px;
  }
}

.emotion-7 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  font-size: 13px;
  line-height: 20px;
  color: #5C6C75;
  margin-bottom: 24px;
}

.emotion-7 strong,
.emotion-7 b {
  font-weight: 600;
}

@media not all and (max-width: 1024px) {
  .emotion-7 {
    font-size: 16px;
    line-height: 28px;
  }
}

.emotion-8 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  font-size: 16px;
  line-height: 28px;
  color: #001E2B;
  margin-bottom: 40px;
  white-space: pre-line;
}

.emotion-8 strong,
.emotion-8 b {
  font-weight: 600;
}

.emotion-9 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  color: #001E2B;
  font-size: 24px;
  line-height: 32px;
  margin-bottom: 24px;
  color: #001E2B;
  font-weight: 500;
}

.emotion-10 {
  display: grid;
  grid-template-columns: 1fr;
  gap: 16px;
}

@media not all and (max-width: 1200px) {
  .emotion-10 {
    grid-template-columns: repeat(2, 1fr);
    gap: 24px;
  }
}

.emotion-11 {
  position: relative;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: border,box-shadow;
  border-radius: 24px;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 13px;
  line-height: 20px;
  padding: 24px;
  min-height: 68px;
  border: 1px solid #E8EDEB;
  box-shadow: 0 4px 10px -4px rgba(0,30,43,0.3);
  background-color: #FFFFFF;
  color: #1C2D38;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  max-width: 472px;
}

.emotion-11:focus {
  outline: none;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC,0 4px 10px -4px rgba(0,30,43,0.3);
}

.emotion-11:hover,
.emotion-11:active {
  border: 1px solid #E8EDEB;
  box-shadow: 0 4px 20px -4px rgba(0,30,43,0.2);
}

.emotion-11:hover:focus,
.emotion-11:active:focus {
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC,0 4px 20px -4px rgba(0,30,43,0.2);
}

.emotion-11:hover {
  -webkit-transform: translateY(-2px);
  -moz-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  margin-bottom: 16px;
  padding: 0 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 6px;
  background-color: #C3E7FE;
}

.emotion-13 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  font-size: 12px;
  line-height: 22px;
  font-weight: 400;
  color: #083C90;
}

.emotion-13 strong,
.emotion-13 b {
  font-weight: 600;
}

.emotion-14 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  color: #001E2B;
  margin: 0;
}

.emotion-14 strong,
.emotion-14 b {
  font-weight: 600;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          href="/products/updates"
        >
          <svg
            aria-label="Arrow Left Icon"
            class="emotion-4"
            fill="none"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 6.832H6.056L7.59 5.297a1 1 0 0 0 0-1.414l-.24-.239a1 1 0 0 0-1.414 0L2.555 7.027q-.014.012-.027.026l-.24.239a1 1 0 0 0 0 1.414l3.652 3.651a1 1 0 0 0 1.414 0l.239-.239a1 1 0 0 0 0-1.414L6.059 9.17H13a1 1 0 0 0 1-1v-.338a1 1 0 0 0-1-1"
              fill="currentColor"
            />
          </svg>
          <p
            class="emotion-5"
          >
            See all updates
          </p>
        </a>
      </div>
      <div>
        <h3
          class="emotion-6"
        >
          MongoDB SQL Interface for Enterprise Advanced Now Generally Available
        </h3>
        <p
          class="emotion-7"
        >
          September 16, 2025
        </p>
      </div>
      <p
        class="emotion-8"
      >
        What it is: 
SQL Interface for Enterprise Advanced provides direct SQL connectivity to self-managed MongoDB deployments through standard ODBC and JDBC connections. It uses the SQL Schema Builder CLI to generate comprehensive JSON schemas and translates SQL queries into MongoDB operations in real-time.

Who it's for:
This is built for data analysts, data scientists, and data engineers who are SQL experts but need access to MongoDB data. It's designed for teams running self-managed MongoDB deployments who want to avoid learning MQL or building custom ETL pipelines.

Why it matters:
SQL Interface eliminates the friction between SQL expertise and MongoDB's document model. Teams can leverage their existing skills and tools to access live MongoDB data without maintaining ETL processes or dealing with outdated information. This enables self-service analytics and faster decision-making while preserving operational workflows.

How to get started:
Check out the link to our documentation below for configuration options and best practices, or head over to the Download Center to get the SQL Schema Builder CLI and drivers for your deployment.
      </p>
      <h3
        class="emotion-9"
      >
        Related Content
      </h3>
      <div
        class="emotion-10"
      >
        <div
          class="emotion-11"
        >
          <div
            class="emotion-12"
          >
            <svg
              aria-label="Note Icon"
              class=""
              color="#083C90"
              fill="none"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 6.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 0 1.5h-3.5a.75.75 0 0 1-.75-.75M4.5 8.75A.75.75 0 0 1 5.25 8h1.5a.75.75 0 0 1 0 1.5h-1.5a.75.75 0 0 1-.75-.75"
                fill="currentColor"
              />
              <path
                clip-rule="evenodd"
                d="m15 10-4 4H3a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2zm-2-6H3v8h7v-2a1 1 0 0 1 1-1h2z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="emotion-13"
            >
              Docs
            </p>
          </div>
          <p
            class="emotion-14"
          />
        </div>
        <div
          class="emotion-11"
        >
          <div
            class="emotion-12"
          >
            <p
              class="emotion-13"
            >
              Web
            </p>
          </div>
          <p
            class="emotion-14"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
